jobbergate-api/jobbergate_api/apps/job_scripts/job_script_files.py

```diff
@@ -0,0 +1,269 @@
+                param_dict_flat[nest_key] = nest_value
+        else:
+            param_dict_flat[key] = value
+    return param_dict_flat
+
+
+def inject_sbatch_params(job_script_data_as_string: str, sbatch_params: List[str]) -> str:
+    """
+    Inject sbatch params into job script.
+
+    Given the job script as job_script_data_as_string, inject the sbatch params in the correct location.
+    """
+    logger.debug("Preparing to inject sbatch params into job script")
+
+    if not sbatch_params:
+        logger.warning("Sbatch param list is empty")
+        return job_script_data_as_string
+
+    first_sbatch_index = job_script_data_as_string.find("#SBATCH")
+    string_slice = job_script_data_as_string[first_sbatch_index:]
+    line_end = string_slice.find("\n") + first_sbatch_index + 1
+
+    inner_string = ""
+    for parameter in sbatch_params:
+        inner_string += "#SBATCH " + parameter + "\n"
+
+    new_job_script_data_as_string = (
+        job_script_data_as_string[:line_end] + inner_string + job_script_data_as_string[line_end:]
+    )
+
+    logger.debug("Done injecting sbatch params into job script")
+    return new_job_script_data_as_string
+
+
+class JobScriptFiles(BaseModel):
+    """
+    Model containing job-script files.
+    """
+
+    main_file_path: Path
+    files: Dict[Path, str]
+
+    @root_validator(pre=False, skip_on_failure=True)
+    def check_main_file_path_is_in_files_keys(cls, values):
+        """
+        Validate the model.
+
+        main_file_path should be found among the files.
+        """
+        if values["main_file_path"] not in values["files"].keys():
+            raise ValueError("main_file_path is not a valid key on the dict files")
+        return values
+
+    @property
+    def main_file(self):
+        """
+        Obtain the main file with this helper property.
+        """
+        return self.files.get(self.main_file_path)
+
+    @classmethod
+    def get_from_s3(cls, job_script_id: int):
+        """
+        Alternative method to initialize the model getting the objects from S3.
+        """
+        logger.debug(f"Getting job-script files from S3: {job_script_id=}")
+        file_manager = cls.file_manager_factory(job_script_id)
+
+        files = {}
+        main_file_path = None
+        main_file_counter = 0
+
+        for s3_path in file_manager.keys():
+            foldername = s3_path.parts[0]
+            dict_path = s3_path.relative_to(foldername)
+            if foldername == JOBSCRIPTS_MAIN_FILE_FOLDER:
+                files[dict_path] = file_manager.get(s3_path)
+                main_file_counter += 1
+                main_file_path = dict_path
+            elif foldername == JOBSCRIPTS_SUPPORTING_FILES_FOLDER:
+                files[dict_path] = file_manager.get(s3_path)
+
+        require_condition(
+            main_file_counter == 1,
+            f"One main file is expected for a job-script, found {main_file_counter}",
```

Comment:
Good to use exact language here like "exactly one" or "one and only one".

Reply:
solved in 9356a000cff69a11ec703749608df83918d937c6